File: 5tip/cpp/Solution1/ConsoleApplication1/Person.cpp
#include <iostream>
#include <ostream>
#include <string>
#include "Person.h"

Person::Person() :age(0) {
	std::cout << "Person created!" << std::endl;
}
Person::Person(std::string firstName, std::string lastName, int age) {
	this->firstName = firstName;
	this->lastName = lastName;
	this->age = age < 0 ? -age : age;
}
Person::~Person() {
	std::cout << "Person destroyed!" << std::endl;
}
int Person::GetAge() const {
	return age;
}
std::string Person::GetFirstName() const {
	return firstName;
}

File: 3tip/3tip_php/cw4/models/Person.php
<?php
class Person
{  // Właściwości klasy Person automatycznie zdefiniowane w konstruktorze
    public function __construct(
        protected int $id = -1,
        protected string $firstName = '',
        protected string $lastName = '',
        protected int $age = 0)
    {
    }
    public function getId(): int
    {
        return $this->id;
    }
    public function getFirstName(): string
    {
        return $this->firstName;
    }
    public function getLastName(): string
    {
        return $this->lastName;
    }
    public function getAge(): int
    {
        return $this->age;
    }
    public function __toString(): string
    {
        return "Imię: " . $this->firstName . ", Nazwisko: "
            . $this->lastName . ", Wiek: " . $this->age;
    }
}
